packages/react/src/drafts/TabPanels/TabPanels.tsx

```diff
@@ -0,0 +1,160 @@
+import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../../utils/polymorphic'
+import clsx from 'clsx'
+import React from 'react'
+import styled from 'styled-components'
+import {get} from '../../constants'
+import {TabContainerElement} from '@github/tab-container-element'
+import {createComponent} from '../../utils/custom-element'
+import sx, {type SxProp} from '../../sx'
+import type {ComponentProps} from '../../utils/types'
+import getGlobalFocusStyles from '../../internal/utils/getGlobalFocusStyles'
+
+const TAB_CLASS = 'TabPanel-tab'
+
+const TabContainer = styled(createComponent(TabContainerElement, 'tab-container'))`
+  & > :not([role='tabpanel']) {
```

Comment:
I'm noticing that the border will flash with our styles since we don't have access to the tablist yet and so it can only span our buttons that are inline-block. Is there any way we could have that node defined?

Reply:
I'd welcome your suggestions on this. I've got as close as I can without adding HTML additional DOM elements. We could add a `<div role="tablist">` pattern but then we'd lose the simplicity of the structure (and the ability to define `<tab><panel><tab><panel>`).

I don't feel super strongly about either of those options but if you can think of any way of doing this with what we've got right now, let me know!

Reply:
Having said that, when you use that alternative structure - the content shift is very noticeable... Perhaps it's best if we _don't_ allow that pattern anyway 🤔 Will discuss with @keithamus

Reply:
Sadly nothing comes to mind on my end @owenniblock 😞 That block level element is definitely what would give that full border effect. From an API perspective, I definitely don't mind having a tab list component and it seems common across other implementations.